src/v/cluster/rm_stm.cc

```diff
@@ -1886,10 +1886,36 @@ void rm_stm::apply_data(model::batch_identity bid, model::offset last_offset) {
     }
 }
 
-void rm_stm::apply_checkpoint([
-  [maybe_unused]] const model::record_batch& batch) {}
+void rm_stm::apply_checkpoint(
+  [[maybe_unused]] const model::record_batch& batch) {}
 
-ss::future<> rm_stm::checkpoint_in_memory_state() const { co_return; }
+ss::future<model::record_batch> rm_stm::make_checkpoint_batch() {
+    iobuf key;
+    co_await serde::write_async(key, _insync_term);
```

Comment:
we should use the same key for the records to allow compaction to take care about the garbage